examples/05_deduplication.py

```diff
@@ -133,46 +120,69 @@
 
 deduplicated_series.plot.barh(figsize=(10, 15))
 plt.xlabel("Medication name")
-plt.ylabel("Counts")
+plt.ylabel("Count")
+plt.show()
 
 ###############################################################################
-# In this example we can correct all spelling mistakes by using the ideal number
-# of clusters as determined by the silhouette score.
+# Here, the silhouette score finds the ideal number of
+# clusters (3) and groups the spelling mistakes.
 #
-# However, often the translation/deduplication won't be perfect and will require some
-# tweaks. In this case, we can construct and update a translation table based on the
-# data returned by |dd|.
-# It consists of the (potentially) misspelled category names as indices and the
-# (potentially) correct categories as values.
+# In practice, the translation/deduplication will often be imperfect
+# and require some tweaks.
+# In this case, we can construct and update a translation table based on the
+# data returned by |deduplicate|.
 
 # create a table that maps original -> corrected categories
 translation_table = pd.Series(deduplicated_data, index=duplicated_names)
 
 # remove duplicates in the original data
 translation_table = translation_table[~translation_table.index.duplicated(keep="first")]
 
+translation_table.head()
+
 ###############################################################################
-# Since the number of correct spellings will likely be much smaller than the
-# number of original categories, we can print the estimated cluster and their
-# most common exemplars (the guessed correct spelling):
+# In this table, we have the category name on the left,
+# and the cluster it was translated to on the right.
+# If we want to adapt the translation table, we can
+# modify it manually.
 
+###############################################################################
+# Visualizing string pair-wise distance between names
+# ---------------------------------------------------
+#
+# Below, we use a heatmap to visualize the pairwise-distance between medication
+# names. A darker color means that two medication names are closer together
+# (i.e. more similar), a lighter color means a larger distance.
+#
 
-def print_corrections(spell_correct):
-    correct = np.unique(spell_correct.values)
-    for c in correct:
-        print(
-            f"Guessed correct spelling: {c!r} for "
-            f"{spell_correct[spell_correct==c].index.values}"
-        )
+from skrub import compute_ngram_distance
+from scipy.spatial.distance import squareform
 
+ngram_distances = compute_ngram_distance(unique_examples)
+square_distances = squareform(ngram_distances)
+
+import seaborn as sns
 
-print_corrections(translation_table)
+fig, axes = plt.subplots(1, 1, figsize=(12, 12))
+sns.heatmap(
+    square_distances, yticklabels=ex_series.index, xticklabels=ex_series.index, ax=axes
+)
+plt.show()
```

Comment:
And this one is a bit too crowded on x, I think it's because the right bar (legend) counts in the allocated space.
Also, I don't think specifying the rows and cols is necessary:
```suggestion
fig, ax = plt.subplots(figsize=(13, 12))
sns.heatmap(
    square_distances, yticklabels=ex_series.index, xticklabels=ex_series.index, ax=ax
)
plt.show()
```